src/main/scala/is/hail/expr/AST.scala

```diff
@@ -829,6 +829,12 @@ case class ApplyMethod(posn: Position, lhs: AST, method: String, args: Array[AST
             case (_: TArray, "map") => ir.ArrayMap(a, name, b)
             case (_: TArray, "filter") => ir.ArrayFilter(a, name, b)
             case (_: TArray, "flatMap") => ir.ArrayFlatMap(a, name, b)
+            case (_: TArray, "exists") =>
+              val v = ir.genUID()
+              ir.ArrayFold(a, ir.False(), v, name, ir.ApplySpecial("||", FastSeq(ir.Ref(v, TBoolean()), b)))
```

Comment:
There's no way to short-circuit right now, correct?

Reply:
not with the IR nodes we currently have, no. (Although it "short-circuits" in the sense that after e.g. the first `true` in an `exists` the ArrayFold operation will stop checking the rest of the elements in the array.)